controllers/spacerequest/spacerequest_controller.go

```diff
@@ -237,9 +239,9 @@ func (r *Reconciler) updateExistingSubSpace(ctx context.Context, spaceRequest *t
 }
 
 // validateNSTemplateTier checks if the provided tierName in the spaceRequest exists and is valid
-func (r *Reconciler) validateNSTemplateTier(ctx context.Context, tierName string) error {
+func (r *Reconciler) validateNSTemplateTier(ctx context.Context, tierName string) (*toolchainv1alpha1.NSTemplateTier, error) {
 	if tierName == "" {
-		return fmt.Errorf("tierName cannot be blank")
+		return &toolchainv1alpha1.NSTemplateTier{}, fmt.Errorf("tierName cannot be blank")
```

Comment:
minor, but we could just return nil here
```suggestion
		return nil, fmt.Errorf("tierName cannot be blank")
```